lib/reek/smells/smell_detector.rb

```diff
@@ -104,16 +93,23 @@ def examine(context)
         self.smells_found += sm
       end
 
-      def enabled_for?(context)
-        enabled? && config_for(context)[SmellConfiguration::ENABLED_KEY] != false
+      def report_on(report)
+        smells_found.each { |smell| smell.report_on(report) }
       end
 
       def exception?(context)
         context.matches?(value(EXCLUDE_KEY, context, DEFAULT_EXCLUDE_SET))
       end
 
-      def report_on(report)
-        smells_found.each { |smell| smell.report_on(report) }
+      protected
+
+      # NOTE: Writers cannot be private in Ruby < 2.2
+      attr_writer :smells_found
```

Comment:
> Those writer work with all ruby versions we run on travis.

Indeed. Something else is going on. The following demonstrates the problem.

```ruby
class Foo
  def initialize
    self.foo = 1
  end
  def bar
    self.foo += 3
  end

  attr_reader :foo

  private

  attr_writer :foo
end

foo = Foo.new
foo.bar
```

This fails on Ruby 2.1 and 2.0. It seems this is due to how the `+=` operator is handled, because it indeed works fine with just straight assignment.

I'll update the note.